File: restaurant-client/src/main/java/com/jdc/restaurant/client/utils/DateDeSerializer.java
package com.jdc.restaurant.client.utils;

import java.io.IOException;
import java.util.Date;

import com.fasterxml.jackson.core.JsonParser;
import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.core.JsonTokenId;
import com.fasterxml.jackson.databind.DeserializationContext;
import com.fasterxml.jackson.databind.JsonDeserializer;

public class DateDeSerializer extends JsonDeserializer<Date>{

	@Override
	public Date deserialize(JsonParser p, DeserializationContext ctxt) throws IOException, JsonProcessingException {
		
		try {
			if(p.getCurrentTokenId() == JsonTokenId.ID_STRING) {
				return DateFormatForJson.df.parse(p.getText());
			} 
			
		} catch (Exception e) {
			e.printStackTrace();
		}
		
		return null;
	}

}

File: restaurant-mobile/app/src/main/java/com/jdc/restaurant/api/utils/DateDeserializer.kt
package com.jdc.restaurant.api.utils

import com.fasterxml.jackson.core.JsonParser
import com.fasterxml.jackson.core.JsonTokenId
import com.fasterxml.jackson.databind.DeserializationContext
import com.fasterxml.jackson.databind.JsonDeserializer
import java.util.*

class DateDeserializer : JsonDeserializer<Date>() {

    override fun deserialize(p: JsonParser?, ctxt: DeserializationContext?): Date? {
        if(null != p && null != ctxt) {
            if(p.currentTokenId == JsonTokenId.ID_STRING) {
                return dateFormat.parse(p.text)
            }
        }

        return null
    }
}